### PR TITLE
add d2l-icon and related tests

### PR DESCRIPTION
**Pull request by jwong-d2l**

### CODE REVIEW

**Comment by tinglisD2L** [comment]
This PR is meant to just add in the icons without much of the styling around it right? (e.g. colours, indenting between icon and text, etc). If so, that's fine. I'll just add more tasks to the story for those changes.

**Comment by jwong-d2l** [comment]
it's intended to just add the icons for now.  I'm currently working on refactoring the CSS for the d2l-activity-link to have a lot less IDs.

File: components/d2l-activity-link.html [hunk 1]
```diff
@@ -55,10 +56,16 @@
 				vertical-align: middle;
 				line-height: 24px;
 			}
+			
+			:host > div > a > d2l-icon {
```
Comment (dbatiste) [inline components/d2l-activity-link.html]: Is it possible to just do:

```
d2l-icon {
   margin-right: ...
}
```

Also, do you need RTL styles to go with?

Comment (tinglisD2L) [reply components/d2l-activity-link.html]: I think our PM said not to worry about RTL for now. Is that right, Joseph?

Comment (jwong-d2l) [reply components/d2l-activity-link.html]: Yes, that's correct, although we should be coding in such a way that our hands aren't tied in regards to rtl later.

**Pull request merged**